packages/cloud_firestore/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/CloudFirestorePlugin.java

```diff
@@ -76,9 +78,16 @@
   private final SparseArray<TaskCompletionSource> completionTasks = new SparseArray<>();
 
   public static void registerWith(PluginRegistry.Registrar registrar) {
-    CloudFirestorePlugin instance = new CloudFirestorePlugin();
+    final CloudFirestorePlugin instance = new CloudFirestorePlugin();
     instance.activity = registrar.activity();
     instance.initInstance(registrar.messenger());
+    registrar.addViewDestroyListener(new PluginRegistry.ViewDestroyListener() {
+      @Override
+      public boolean onViewDestroy(FlutterNativeView view) {
+        instance.removeSnapshotListeners();
```

Comment:
Maybe treat this one as `onDetachedFromEngine` and do all cleanup tasks? @matthew-carroll is that a good assumption?

Reply:
@ened are you thinking about cleaning up the channel?

Reply:
@kroikie it seems good practice (& is part of the V2 generated code), so yes. 

IMHO, this shouldn't be needed in the long run, as a closing engine should clean everything and take everything with it.

Reply:
Agreed